### PR TITLE
Update sidekicks

### PR DESCRIPTION
**Pull request by tamas-molnar**
* templated, switched to VCB, new port setting pattern
 * fleet-unit-hc
* new pattern for
 * pub-varnish
 * agg-hc
 * system-hc
* tested in pub-xp, looks fine
* stopped, then started all machines, and no HC issues
* manual actions after deploy to prod
 * `fleetctl destroy fleet-unit-healthcheck-sidekick.service`
 * `fleetctl destroy fleet-unit-healthcheck.service`
* etcd values cleanup - delete everything matched by `etcdctl ls / --recursive | grep "fleet-unit-healthcheck$\|fleet-unit-healthcheck/"`
* will only merge to prod after the freeze

### CODE REVIEW

File: pub-varnish-sidekick@.service [hunk 1]
```diff
@@ -4,12 +4,14 @@ BindsTo=pub-varnish@%i.service
...
-    PORT=`[ $PORT ] && echo $PORT || echo $(/usr/bin/docker port varnish-%i 80 | cut -d":" -f2)`; \
-    etcdctl set /ft/services/varnish/servers/%i http://%H:$PORT --ttl 5; \
-    sleep 4; \
-  done'
+RemainAfterExit=yes
+ExecStart=/bin/sh -c ""\
+  export SERVICE=$(echo %p | sed 's/-sidekick//g'); \
+  while [ -z $PORT ]; do \
+     sleep 5; \
+     PORT=`echo $(/usr/bin/docker port $SERVICE-%i 8080 | cut -d':' -f2)`; \
```
Comment (catalin-ilea) [inline pub-varnish-sidekick@.service]: Should be 80 instead of 8080

File: pub-varnish-sidekick@.service [hunk 2]
```diff
@@ -4,12 +4,14 @@ BindsTo=pub-varnish@%i.service
...
 
 [Service]
-ExecStart=/bin/sh -c '\
-  while true; do \
-    PORT=`[ $PORT ] && echo $PORT || echo $(/usr/bin/docker port varnish-%i 80 | cut -d":" -f2)`; \
-    etcdctl set /ft/services/varnish/servers/%i http://%H:$PORT --ttl 5; \
-    sleep 4; \
-  done'
+RemainAfterExit=yes
+ExecStart=/bin/sh -c ""\
```
Comment (catalin-ilea) [inline pub-varnish-sidekick@.service]: multiple "

File: services.yaml [hunk 3]
```diff
@@ -58,8 +60,10 @@ services:
   version: v1.0.2
   count: 2
 - name: splunk-forwarder.service
-- name: system-healthcheck-sidekick.service
-- name: system-healthcheck.service
+- name: system-healthcheck-sidekick@.service
```
Comment (catalin-ilea) [inline services.yaml]: This should be a global service. Options are leave it non-templated(VCB can't handle it) or hardcode the count to the no of ec2 instances in the cluster(not very nice).
@martingartonft thoughts?

Comment (catalin-ilea) [inline services.yaml]: We could agree that all non templated services that we add need to be global and update VCB to deal with this.

File: aggregate-healthcheck-sidekick@.service [hunk 4]
```diff
@@ -4,22 +4,23 @@ BindsTo=aggregate-healthcheck@%i.service
...
+RemainAfterExit=yes
+ExecStart=/bin/sh -c "\
+  export SERVICE=$(echo %p | sed 's/-sidekick//g'); \
+  etcdctl mkdir /ft/services/$SERVICE/servers; \
+  etcdctl set /vulcand/frontends/aggregate-healthcheck-public/frontend '{\"Type\":\"http\", \"BackendId\":\"vcb-aggregate-healthcheck\", \"Route\":\"Path(`/__health`)\"}'; \
+  etcdctl set /vulcand/frontends/aggregate-healthcheck-public-gtg/frontend '{\"Type\":\"http\", \"BackendId\":\"vcb-aggregate-healthcheck\", \"Route\":\"Path(`/__gtg`)\"}'; \
   etcdctl set /ft/healthcheck-categories/read/period_seconds 4; \
   etcdctl set /ft/healthcheck-categories/read/is_resilient true; \
   etcdctl set /ft/healthcheck-categories/system/period_seconds 60; \
-  etcdctl set /ft/healthcheck-categories/system/is_resilient false"
```
Comment (catalin-ilea) [inline aggregate-healthcheck-sidekick@.service]: Don't think you should remove this line

File: system-healthcheck-sidekick@.service [hunk 5]
```diff
@@ -1,22 +1,24 @@
...
-  while true; do \
-    PORT=`[ $PORT ] && echo $PORT || echo $(/usr/bin/docker port system-healthcheck 8080 | cut -d':' -f2)`; \
-    etcdctl set /vulcand/backends/system-healthcheck-%H/servers/srv%H \"{\\\"url\\\": \\\"http://$HOST_IP:$PORT\\\"}\" --ttl 600; \
-    sleep 120; \
-  done"
+  while [ -z $PORT ]; do \
+    sleep 5; \
+    PORT=`echo $(/usr/bin/docker port $SERVICE-%i 8080 | cut -d':' -f2)`; \
+  done; \
+  etcdctl set /vulcand/backends/system-healthcheck-%H/servers/srv%H \"{\\\"url\\\": \\\"http://$HOST_IP:$PORT\\\"}\" --ttl 600;"
```
Comment (Nesze) [inline system-healthcheck-sidekick@.service]: I guess the 'ttl' option should be removed.

File: aggregate-healthcheck-sidekick@.service (hunk 4 above)
Comment (tamas-molnar) [inline aggregate-healthcheck-sidekick@.service]: @catalin-ilea  It's still there, just a bit lower https://github.com/Financial-Times/pub-service-files/pull/7/files#diff-75564dd80b61174c8fd7a8fceb394208R16

File: services.yaml (hunk 3 above)
Comment (tamas-molnar) [inline services.yaml]: I only templated this to use VCB, then noticed I can't do that, I'll revert the templating.

**Comment by catalin-ilea** [comment]
lgtm

**Comment by tosan88** [comment]
:+1:

**Comment by kovacshuni** [comment]
+1

**Pull request merged**